src/nvim/ops.c

```diff
@@ -5732,3 +5738,19 @@ const yankreg_T *op_register_get(const char name)
   }
   return &y_regs[i];
 }
+
+/// Set the previous yank register
+///
+/// @param[in]  name  Register name.
+///
+/// @return true on success, false on failure.
+bool op_register_set_previous(const char name)
```

Comment:
Also, it isn't a pure function. It modifies the global variable `y_previous`. When I compiled with `FUNC_ATTR_PURE` it never set `y_previous`. In fact, in gdb it just skipped over calling the function (since I wasn't using the result the compiler probably optimized it away. No reason to call a pure function if you aren't going to use it's return value after all).